### ISSUE TITLE
V7.3.2 out of scope?

### ISSUE THREAD

**Issue by elarlang** [OWASP/ASVS#878]
[7.3.2](https://github.com/OWASP/ASVS/blob/master/4.0/en/0x15-V7-Error-Logging.md#v73-log-protection-requirements)

> V7.3.2 Verify that all events are protected from injection when viewed in log viewing software.

How one web application can protect some external log viewing software against injection attacks? It can not.

It would be the same wrong like "encode data to HTML specialcharters to database then you don't have XSS/HTML injection when you display it" - it means, it does not make sense.

If web application displays logs and there are injection on displaying them, it's more output encoding problem (like requirement V5.3.3), if it's outside of the application, it's also outside of testing scope.

To have proper encoding in place, it's covered with V7.3.1:

> Verify that the application appropriately encodes user-supplied data to prevent log injection.

**Comment by Sjord** [comment]
I agree if this applies to HTML injection. Sanitising input before logging it does not make sense. If someone submits `<script>alert(1)` in a form, you want to exactly that to show up in the log, in order to investigate attacks.

However, the application should protect against log injection. If lines are logged into a plain text file, newlines in user input should be escaped by the application before writing it to the log, to prevent attackers from injecting new log lines.

**Comment by elarlang** [comment]
> I agree if this applies to HTML injection. Sanitising input before logging it does not make sense. If someone submits `<script>alert(1)` in a form, you want to exactly that to show up in the log, in order to investigate attacks.

Like I already wrote, for that you have requirement V5.3.3

> However, the application should protect against log injection. If lines are logged into a plain text file, newlines in user input should be escaped by the application before writing it to the log, to prevent attackers from injecting new log lines.

Like I already wrote, for that you have requirement V7.3.1

**Comment by Sjord** [comment]
> Like I already wrote ... 

I feel like you are a bit annoyed with my reaction, is that the case? I feel like your reaction is a little bit blunt, and I don't understand why you feel the need to repeat yourself. It was not by intention to argue with you. I was trying to add to the conversation.

**Comment by elarlang** [comment]
Sorry, no intent to cause negative reactions, my comments corrected. As my English is not the best one, I'm not always sure I'm I understood the way like I want. That's why this repeating.

**Comment by jmanico** [comment]
Elar is sharp and not always polite via western sensibilities. Just fight back he can take it. :) But please note that Elar is one of the hardest working OWASP volunteers and I know he’s very intelligent and his contributions are really important to the foundation.

There are plenty of older logging abstractions and languages where attackers can abuse logs, delete log messages add new log messages and other kinds of abuse and this is called log injection it’s an older vulnerability class that in the past required extra work to stop. This is not about XSS, It’s mostly about adding new malicious logging entries or defacing login entries with user input. 

https://owasp.org/www-community/attacks/Log_Injection

If you’re not doing proper logging Security and I can start adding my own logging events I can undermine the records of transactions in your log. This matters. But I’m not sure if developers need to take corrective action or if the login classes of the day handle this problem.

--
Jim Manico
@Manicode

**Comment by elarlang** [comment]
@jmanico - from your description point of view, I would say that V7.3.1 and V7.3.2 are duplicates:
* V7.3.1 Verify that the application appropriately encodes user-supplied data to prevent log injection.
* V7.3.2 Verify that all events are protected from injection when viewed in log viewing software.

The main goal is to avoid log injection - it should be covered already with V7.3.1. You can not insert new lines, right-to-left, special control characters for terminals etc. It's all covered here.

In general or with "over-applications" view I can understand need for requirement V7.3.2, but from testing one application point of view, it's out of scope. If problem happens in next component, it's not anymore under web application control and can not be fixed with web application.

**Comment by jmanico** [comment]
I agree with you 100%. Good call, Elar.

--
Jim Manico
@Manicode

**Comment by elarlang** [comment]
PR after #893 merge

**Comment by jmanico** [comment]
#893 is merged, lets see that PR! :)
